### ISSUE TITLE
Detect references to invokeDynamic using method handles to forbidden methods

### ISSUE THREAD

**Issue by GoogleCodeExporter** [policeman-tools/forbidden-apis#11]
```
Currently the following Java 8 call does not get detected as forbidden call:

  Arrays.sort(someArray, Float::compare);

if the forbidden API signatures contains:

  java.lang.Float#compare(float,float)

The autoboxing invoved here is not the problem, the problem is more the Java 7 
feature of an invokeDynamic to a dynamic call site using a method handle to a 
static or virtual method which is on the forbidden list.

In short: we need detection for this crazy new Java 8 syntax:

  Class::method

using the new Java 7 MethodHandle, exposed by this ASM 4.0 class: 
http://asm.ow2.org/asm40/javadoc/user/org/objectweb/asm/Handle.html

The MethodVisitor has to detect invokeDynamic using a handle to our forbidden 
method.
```

Original issue reported on code.google.com by `uwe.h.schindler` on 26 Sep 2013 at 1:30
